qiskit-superstaq/qiskit_superstaq/superstaq_backend.py

```diff
@@ -37,63 +39,84 @@ def __init__(self, provider: qss.SuperstaqProvider, target: str) -> None:
             target: A string containing the name of a target backend.
         """
         gss.validation.validate_target(target)
+        description = f"Superstaq '{target}' backend"
 
-        self._configuration: qiskit.providers.models.BackendConfiguration | None
-        self._provider = provider
-        self._target = target
+        self._target_info: dict[str, object] | None = None
 
-        super().__init__(None, provider=provider)
+        super().__init__(provider=provider, name=target, description=description)
 
     @classmethod
     def _default_options(cls) -> qiskit.providers.Options:
         return qiskit.providers.Options(shots=1000)
 
-    def name(self) -> str:
-        """Gets the target name with which this backend is associated.
-
-        Returns:
-            The target name associated with this backend.
-        """
-        return self._target
-
     def __eq__(self, other: Any) -> bool:
         if not isinstance(other, qss.SuperstaqBackend):
             return False
 
-        return self._provider == other._provider and self.configuration() == other.configuration()
+        return self._provider == other._provider and self.target_info() == other.target_info()
 
-    def configuration(self) -> qiskit.providers.models.BackendConfiguration:
+    def configuration(self) -> BackendConfiguration:
         """Retrieves configuration information for this target.
 
         Returns:
             A backend configuration object containing various hardware parameters.
         """
-        if self._configuration is None:
-            target_info = self._provider._client.target_info(self._target)["target_info"]
-
-            num_qubits = target_info.pop("num_qubits", None)
-            configuration_dict = {
-                "backend_name": target_info.pop("target", None),
-                "basis_gates": target_info.pop("native_gate_set", []),
-                "backend_version": "n/a",
-                "n_qubits": num_qubits,
-                "gates": [],
-                "local": False,
-                "simulator": False,
-                "conditional": False,
-                "open_pulse": False,
-                "memory": False,
-                "max_shots": None,
-                "coupling_map": None,
-                "description": f"{num_qubits} qubit device",
-            }
+        warnings.warn(
+            "The `.configuration()` method of `SuperstaqBackend` has been deprecated, and will be "
+            "removed in a future version of qiskit-superstaq. Instead, use attributes of the "
+            "backend itself (e.g. `backend.num_qubits`), or of its `.target` attribute.",
+            DeprecationWarning,
+        )
```

Comment:
Calling `backend.configuration()` didn't raise a warning for me locally, so maybe: 
```suggestion
            DeprecationWarning,
            stacklevel=2,
        )
```